src/slam_toolbox_common.cpp

```diff
@@ -522,7 +523,10 @@ bool SlamToolbox::shouldProcessScan(
 
   // check moved enough, within 10% for correction error
   const double dist2 = last_pose.SquaredDistance(pose);
-  if (dist2 < 0.8 * min_dist2 || scan_ctr < 5) {
+  double heading_diff = pose.GetHeading() - last_pose.GetHeading();
+  if (heading_diff < 0)
+      heading_diff = -heading_diff;
+  if ((dist2 < 0.8 * min_dist2 && heading_diff < 0.9 * min_heading) || scan_ctr < 5) {
```

Comment:
Why 0.9? That seems arbitrary

Reply:
The comment a few lines above said:
```c
// check moved enough, within 10% for correction error
```
so I followed that by multiplying 0.9.